### PR TITLE
Adjust grpc defaults for improved performance

### PR DESCRIPTION
**Pull request by deadeyegoodwin**

### CODE REVIEW

File: src/servers/main.cc [hunk 1]
```diff
@@ -106,18 +106,17 @@ int32_t trace_rate_ = 1000;
...
+int grpc_stream_infer_thread_cnt_ = 1;
 
 // The maximum number of inference request/response objects that
 // remain allocated for reuse. As long as the number of in-flight
 // requests doesn't exceed this value there will be no
-// allocation/deallocation of request/response objects. Higher values
-// trade-off increased memory usage for higher performance.
-int grpc_infer_allocation_pool_size_ = 128;
+// allocation/deallocation of request/response objects.
+int grpc_infer_allocation_pool_size_ = 0;
```
Comment (GuanLuo) [inline src/servers/main.cc]: So reusing the inference object actually has negative impact on performance?

File: src/servers/grpc_server.cc [hunk 2]
```diff
@@ -340,14 +338,17 @@ class Handler : public GRPCServer::HandlerBase {
 
   void StateRelease(State* state)
   {
-    std::unique_lock<std::mutex> lock(alloc_mu_);
+    if (max_state_bucket_count_ > 0) {
+      std::unique_lock<std::mutex> lock(alloc_mu_);
```
Comment (GuanLuo) [inline src/servers/grpc_server.cc]: Can just use `lock_guard` for scoped lock

File: src/servers/main.cc (hunk 1 above)
Comment (deadeyegoodwin) [reply src/servers/main.cc]: For the case I'm testing. Maybe not for request with large input/output tensors. We may need to do something dynamic based on the size of the message.

File: src/servers/grpc_server.cc (hunk 2 above)
Comment (deadeyegoodwin) [reply src/servers/grpc_server.cc]: Fixed

**Pull request merged**